mongodb/src/main/java/org/apache/metamodel/mongodb/MongoDbDataContext.java

```diff
@@ -435,6 +442,19 @@ private String getOperatorName(FilterItem item) {
         return operatorName;
     }
 
+    private Pattern turnOperandIntoRegExp(Object operand) {
+        String operandAsString = operand.toString();
+        StringBuilder operandAsRegExp = new StringBuilder(replaceWildCardLikeChars(operandAsString));
+        if (!operandAsString.startsWith("%") && operandAsString.contains("%")) {
+            operandAsRegExp = operandAsRegExp.insert(0, "^");
+        }
+        return Pattern.compile(operandAsRegExp.toString(), Pattern.CASE_INSENSITIVE);
+    }
```

Comment:
Isn't this missing the ```LIKE '%xxxxxx'``` case? I'd imagine a regex with $ in the end in that case, so that ```xxxxxxy``` wouldn't match, while ```yxxxxxx``` would.

But do we even need all these special cases? I would guess that always adding ^ and $ would always make it work? A few examples:
```LIKE 'xxxxxx'``` would become ```/^xxxxxxx$/``` which would only match ```xxxxxx```.
```LIKE '%xxxxxx'``` would become ```/^.*xxxxxx$/``` which would match anything ending with ```xxxxxx```.
```LIKE 'xxxxxx%'``` would become ```/^xxxxxx.*$/``` which would match anything starting with ```xxxxxx```.